### ISSUE TITLE
No "./autogen.sh" during installation

### ISSUE THREAD

**Issue by xushanthu-2014** [rocky/remake#118]
Hi I want to install remake in my linux server to debug the gnu make, and I simply follows the following commands to download remake from sourceforge and then install it as what I was told in https://remake.readthedocs.io/en/latest/install.html#sourceforge:
```
wget https://sourceforge.net/projects/bashdb/files/remake/4.2+dbg-1.5/remake-4.2.1+dbg-1.5.tar.bz2
tar -xpf remake-4.2.1+dbg-1.5.tar.bz2 
cd remake-4.2.1+dbg-1.5/
./autogen.sh
```
Then I found there is no autogen.sh in this directory. So is there something wrong? Or I missed something?
Thanks and hope for your early reply!

**Comment by rocky** [comment]
I found that there is an `INSTALL-from-source.md` in this directory. Did you look at that?

**Comment by martencassel** [comment]
I'm also trying to setup remake using https://github.com/rocky/remake/blob/remake-4-3/doc/readthedocs/install.rst#sourceforge
I perform 

```bash
docker run -it --rm fedora:35 /bin/bash
#
# dnf -y update
# dnf -y install @development-tools
# dnf -y install autoconf automake gettext-devel gettext readline-devel wget
# mkdir /root/src && cd /root/src
# wget https://sourceforge.net/projects/bashdb/files/remake/4.3%2Bdbg-1.5/remake-4.3%2Bdbg-1.5.tar.gz/download -O remake-4.3_dbg-1.5.tar.gz
# tar xvf remake-4.3_dbg-1.5.tar.gz
# cd /root/src/remake-4.3+dbg-1.5
# ./autogen.sh
bash: ./autogen.sh: No such file or directory
```
I see that this file is available in the [github repo](https://github.com/rocky/remake/blob/remake-4-3/autogen.sh) but seems not to be available in the sourceforce archive.

**Comment by rocky** [comment]
Distributed files (what is in tarballs) and development files (what is in the github) are different although, of course, they are very similar - to the point where people think you can use one for the other too often. 

What is in github and in development do not contain any derived files. For example, language-specific translation files for messages are not in the github repo but are in the tarball . 

What is not in tarballs is any code that is intermediate to building like autogen.sh. In the tarball, you use `configure` only, not the code that is used to create `configure`.

In short, if you find you need to use `autogen.sh` then you should be pulling from github, and if you find you don't want to go through that elaborate process but just want a `configure` script that works and want something which is known to be stable, then use the tarballs. 

BTW, I believe github provides a way to get a tarball or zip created on the fly from the current github repo.
